lib/puppet/rest/client.rb

```diff
@@ -54,6 +54,15 @@ def get(url, query: nil, header: nil, &block)
       end
     end
 
+    # Make a PUT request to the specified URL with the specified params.
+    # @param [String] url the full path to query
+    # @param [String/Hash] body the contents of the PUT request
+    # @param [Hash] query any URL params to add to send to the endpoint
+    # @param [Hash] header any additional entries to add to the default header
+    def put(url, body:, query: nil, header: nil)
```

Comment:
It's actually required this way too (since no value is specified). I went back and forth about whether to label all of them, including URL, since it's a bit weird having this mix, though I worry that will get too verbose. I do like keeping labels on all of the parts you can send though, which makes them order-independent.